fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -1197,26 +1198,48 @@ void insertValueBuggy() {
 
     @Test
     void markReadable() throws Exception {
-        final String indexName = "MySimpleRecord$str_value_indexed";
+        try (FDBRecordContext context = openContext()) {
+            final RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestNoIndexesProto.getDescriptor());
+            recordStore = FDBRecordStore.newBuilder().setContext(context).setMetaDataProvider(builder).setKeySpacePath(path).createOrOpen();
+            // Save 20 records without any indexes.
+            for (int i = 0; i < 20; i++) {
+                TestNoIndexesProto.MySimpleRecord record = TestNoIndexesProto.MySimpleRecord.newBuilder()
+                        .setRecNo(i).setNumValue(i + 1000).build();
+                recordStore.saveRecord(record);
+            }
+            commit(context);
+        }
 
+        final String indexName = "MySimpleRecord$str_value_indexed";
         try (FDBRecordContext context = openContext()) {
-            uncheckedOpenSimpleRecordStore(context);
+            openSimpleRecordStore(context);
             assertThat(recordStore.isIndexWriteOnly(indexName), is(false));
             recordStore.clearAndMarkIndexWriteOnly(indexName).get();
             assertThat(recordStore.isIndexWriteOnly(indexName), is(true));
-            context.commit();
+            commit(context);
         }
 
+        final String throwMsg = "Intentionally thrown during test";
+        final AtomicLong counter = new AtomicLong(0);
+        UnaryOperator<OnlineIndexOperationConfig> configLoader = old -> {
+            if (counter.incrementAndGet() > 1) {
```

Comment:
This is a bit confusing, perhaps a comment that the intent is to fail the index build after completing 1 transaction would make it clearer, or putting it in a method.

Reply:
✅